api/client/client_test.go

```diff
@@ -0,0 +1,487 @@
+	if err != nil {
+		t.Fatalf("failed to start: %v", err)
+	}
+}
+
+func TestStartFromAddress(t *testing.T) {
+	_, clean := server(t)
+	defer clean()
+	client := client(t)
+
+	storeID, _ := client.NewStore()
+	err := client.RegisterSchema(storeID, modelName, schema)
+	checkErr(t, err)
+
+	// TODO: figure out how to test this
+	// client.StartFromAddress(storeId, <multiaddress>, <read key>, <follow key>)
+}
+
+func TestModelCreate(t *testing.T) {
+	_, clean := server(t)
+	defer clean()
+	client := client(t)
+
+	storeID, _ := client.NewStore()
+	err := client.RegisterSchema(storeID, modelName, schema)
+	checkErr(t, err)
+	err = client.Start(storeID)
+	checkErr(t, err)
+
+	err = client.ModelCreate(storeID, modelName, createPerson())
+	if err != nil {
+		t.Fatalf("failed to create model: %v", err)
+	}
+}
+
+func TestModelSave(t *testing.T) {
+	_, clean := server(t)
+	defer clean()
+	client := client(t)
+
+	storeID, _ := client.NewStore()
+	err := client.RegisterSchema(storeID, modelName, schema)
+	checkErr(t, err)
+	err = client.Start(storeID)
+	checkErr(t, err)
+
+	person := createPerson()
+
+	err = client.ModelCreate(storeID, modelName, person)
+	checkErr(t, err)
+
+	person.Age = 30
+	err = client.ModelSave(storeID, modelName, person)
+	if err != nil {
+		t.Fatalf("failed to save model: %v", err)
+	}
+}
+
+func TestModelDelete(t *testing.T) {
+	_, clean := server(t)
+	defer clean()
+	client := client(t)
+
+	storeID, _ := client.NewStore()
+	err := client.RegisterSchema(storeID, modelName, schema)
+	checkErr(t, err)
+	err = client.Start(storeID)
+	checkErr(t, err)
+
+	person := createPerson()
+
+	err = client.ModelCreate(storeID, modelName, person)
+	checkErr(t, err)
+
+	err = client.ModelDelete(storeID, modelName, person.ID)
+	if err != nil {
+		t.Fatalf("failed to delete model: %v", err)
+	}
+}
+
+func TestModelHas(t *testing.T) {
+	_, clean := server(t)
+	defer clean()
+	client := client(t)
+
+	storeID, _ := client.NewStore()
+	err := client.RegisterSchema(storeID, modelName, schema)
+	checkErr(t, err)
+	err = client.Start(storeID)
+	checkErr(t, err)
+
+	person := createPerson()
+
+	err = client.ModelCreate(storeID, modelName, person)
+	checkErr(t, err)
+
+	exists, err := client.ModelHas(storeID, modelName, person.ID)
+	if err != nil {
+		t.Fatalf("failed to check model has: %v", err)
+	}
+	if !exists {
+		t.Fatal("model should exist but it doesn't")
+	}
+}
+
+func TestModelFind(t *testing.T) {
+	_, clean := server(t)
+	defer clean()
+	client := client(t)
+
+	storeID, _ := client.NewStore()
+	err := client.RegisterSchema(storeID, modelName, schema)
+	checkErr(t, err)
+	err = client.Start(storeID)
+	checkErr(t, err)
+
+	person := createPerson()
+
+	err = client.ModelCreate(storeID, modelName, person)
+	checkErr(t, err)
+
+	q := es.JSONQuery{
+		Ands: []es.JSONCriterion{
+			es.JSONCriterion{
+				FieldPath: "lastName",
+				Operation: es.Eq,
+				Value: es.JSONValue{
+					String: &person.LastName,
+				},
+			},
+		},
+	}
```

Comment:
I haven't been able to find how to create a `JSONQuery` using a fluent API. Can someone point me to that?

Reply:
Sure, should be something analogous to [this](https://github.com/textileio/go-textile-threads/blob/master/eventstore/query.go#L30) and [this](https://github.com/textileio/go-textile-threads/blob/master/eventstore/criterion.go#L57)

Reply:
Hmm I think I'm just not making some connection here... Those links point to functions that operate on the non-json queries and criterions. I need a similar API for JSONQuery, right? Does it exist?

Reply:
I think in this case (with JSON mode), the "fluent" part has to be implemtned on the client.

Reply:
Ah ok.

Reply:
Issue here https://github.com/textileio/go-textile-threads/issues/145